File: CLRDEV9/CLRDEV9/PSE/CLR_PSE_PluginLog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PSE
{
    public class CLR_PSE_PluginLog
    {
        public CLR_PSE_Callbacks.Log_Open Open;
        public CLR_PSE_Callbacks.Close Close;
        public CLR_PSE_Callbacks.Log_Write LogWrite;

        public CLR_PSE_Callbacks.Log_Write LogWriteLine;
        public CLR_PSE_Callbacks.Log_SetValue SetWriteToFile;

        public void ErrorWrite(string str)
        {
            Console.Error.Write(str);
            //log to stderr
            this.LogWrite(str);
        }
        public void ErrorWriteLine(string str)
        {
            Console.Error.WriteLine(str);
            //log to stderr
            this.LogWriteLine(str);
        }

    }
}

File: CLRUSB/CLRUSB/PSE/CLR_PSE_PluginLog.vb
Imports System.IO
Namespace Global.PSE
    Public Class CLR_PSE_PluginLog
        Public Open As CLR_PSE_Callbacks.Log_Open
        Public Close As CLR_PSE_Callbacks.Close
        Public LogWrite As CLR_PSE_Callbacks.Log_Write
        Public LogWriteLine As CLR_PSE_Callbacks.Log_Write

        Public SetWriteToFile As CLR_PSE_Callbacks.Log_SetValue
        'Public SetWriteToConsole As CLR_Callbacks.Log_SetValue

        Public Sub ErrorWrite(str As String)
            Console.Error.Write(str) 'log to stderr
            Me.LogWrite(str)
        End Sub
        Public Sub ErrorWriteLine(str As String)
            Console.Error.WriteLine(str) 'log to stderr
            Me.LogWriteLine(str)
        End Sub

    End Class
End Namespace
